fe1-web/src/features/evoting/network/__tests__/ElectionMessageApi.test.ts

```diff
@@ -0,0 +1,172 @@
+import { publish } from 'core/network';
+import { channelFromIds, Timestamp } from 'core/objects';
+import { SelectedBallots } from 'features/evoting/objects';
+import {
+  mockElectionNotStarted,
+  mockElectionOpened,
+} from 'features/evoting/objects/__tests__/utils';
+import 'jest-extended';
+import { mockLaoIdHash } from '__tests__/utils';
+import {
+  castVote,
+  openElection,
+  requestCreateElection,
+  terminateElection,
+} from '../ElectionMessageApi';
+import { CastVote, EndElection, SetupElection } from '../messages';
+import { OpenElection } from '../messages/OpenElection';
+
+const mockChannelId = 'some channel id';
+
+jest.mock('core/objects', () => {
+  return {
+    ...jest.requireActual('core/objects'),
+    channelFromIds: jest.fn().mockImplementation(() => mockChannelId),
+  };
+});
+
+jest.mock('core/network', () => {
+  return {
+    ...jest.requireActual('core/network'),
+    publish: jest.fn().mockImplementation(() => Promise.resolve()),
+  };
+});
+
+afterEach(() => {
+  (channelFromIds as jest.Mock).mockClear();
+  (publish as jest.Mock).mockClear();
+});
+afterAll(() => {
+  (channelFromIds as jest.Mock).mockReset();
+  (publish as jest.Mock).mockReset();
+});
+
+describe('mockElectionNotStarted.id', () => {
+  describe('requestCreateElection', () => {
+    it('works as expected using a valid set of parameters', () => {
+      requestCreateElection(
+        mockLaoIdHash,
+        mockElectionNotStarted.name,
+        mockElectionNotStarted.version,
+        mockElectionNotStarted.start,
+        mockElectionNotStarted.end,
+        mockElectionNotStarted.questions,
+        mockElectionNotStarted.createdAt,
+      );
+
+      expect(channelFromIds).toHaveBeenCalledWith(mockLaoIdHash);
+      expect(channelFromIds).toHaveBeenCalledTimes(1);
+
+      expect((publish as jest.Mock).mock.calls[0][0]).toEqual(mockChannelId);
```

Comment:
you could define a `const publishMock = (publish as jest.Mock).mock` and use that everywhere here